lib/rucio/core/replica.py

```diff
@@ -835,6 +836,12 @@ def _list_replicas(dataset_clause, file_clause, state_clause, show_pfns, schemes
                                     if root_proxy_internal and 'value' in root_proxy_internal[0]:
                                         pfn = root_proxy_internal[0]['value'] + '//' + pfn
 
+                        # do we need to sign the URLs?
+                        if sign_urls and protocol.attributes['scheme'] == 'https':
+                            sign = get_rse_attribute('sign_url_gcs', rse_info[rse]['id'], session=session)
+                            if sign and isinstance(sign, list) and sign[0]:
+                                pfn = get_signed_url(service='gcs', operation='read', url=pfn)
+
```

Comment:
I have the same concerns about the performance by having these calls to third party services here in case something goes bad: long duration db sessions, impact on the pools, timeout and the corresponding response time increase, earth quake, tornado, etc. Any ideas on how to address these concerns ? you can do better :)

Reply:
there is no call to third party services. the signature is done rucio side. when i tested it, i managed 10k signatures per seconds on my laptop. i don't think this is an issue.

Reply:
That's like macaroon or scitoken which is good but the issue remains for s3 and swift unfortunately.

Reply:
Yep, these should not be done in here.

Reply:
yes ... but it's the same call and functionality.. going back to original comment ;)